src/runmode-pfring.c

```diff
@@ -290,31 +291,9 @@ static void *ParsePfringConfig(const char *iface)
     }
 
     /*load pfring bpf filter*/
-    /* command line value has precedence */
-    if (ConfGet("bpf-filter", &bpf_filter) == 1) {
-        if (strlen(bpf_filter) > 0) {
-            pfconf->bpf_filter = SCStrdup(bpf_filter);
-            if (unlikely(pfconf->bpf_filter == NULL)) {
-                SCLogError(SC_ERR_MEM_ALLOC,
-                           "Can't allocate BPF filter string");
-            } else {
-                SCLogDebug("Going to use command-line provided bpf filter %s",
-                           pfconf->bpf_filter);
-            }
-        }
-    } else {
-        if (ConfGetChildValueWithDefault(if_root, if_default, "bpf-filter", &bpf_filter) == 1) {
-            if (strlen(bpf_filter) > 0) {
-                pfconf->bpf_filter = SCStrdup(bpf_filter);
-                if (unlikely(pfconf->bpf_filter == NULL)) {
-                    SCLogError(SC_ERR_MEM_ALLOC,
-                               "Can't allocate BPF filter string");
-                } else {
-                    SCLogDebug("Going to use bpf filter %s",
-                               pfconf->bpf_filter);
-                }
-            }
-        }
+    if (ParseBpfConfig(if_root, &tmpbpf) == 1) {
```

Comment:
This seems to remove the `default` logic. The goal there is to make this work:
```yaml
pfring:
  - interface: eth0
  - interface: eth1
  - interface: default
    bpf-filter: tcp
```
If `bpf-filter` is not present in on of the interface specific yaml blocks, it uses the one from `default`.